File: Singleton.kt
object Singleton{
    var singleton: Singleton? = null
    init{
        println("Object created!")
    }
    fun createSingleton(): Singleton {
        
        if( this.singleton == null ){
            this.singleton = Singleton
        }else{
            print("Already created")
        }
        return this.singleton!!
    }
}

fun main() {
        Singleton.createSingleton()
        Singleton.createSingleton()
}

File: Singleton.ts
class Singleton{
    
    private static singleton:Singleton;
    
    private constructor(){
        console.log('created');
    }

    static createSingleton():Singleton{
        if( Singleton.singleton == undefined ){
            Singleton.singleton = new Singleton();
            return Singleton.singleton;
        }else{
            console.log('Already defined');
        }
        return Singleton.singleton;
    }
}
const p1 = Singleton.createSingleton();
const p2 = Singleton.createSingleton();